extensions/amp-truncate-text/1.0/storybook/Basic.amp.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {number, withKnobs} from '@storybook/addon-knobs';
+import {withA11y} from '@storybook/addon-a11y';
+import {withAmp} from '@ampproject/storybook-addon';
+
+export default {
+  title: 'amp-truncate-text-1_0',
+  decorators: [withKnobs, withA11y, withAmp],
+
+  parameters: {
+    extensions: [
+      {
+        name: 'amp-truncate-text',
+        version: '1.0',
+      },
+    ],
+    experiments: ['bento'],
+  },
+};
+
+export const _default = () => {
+  const width = number('width (em)', 20);
+  const height = number('height (em)', 4);
+
+  return (
+    <amp-truncate-text
+      layout="fixed"
+      className="box"
+      height={`${height}em`}
+      width={`${width}em`}
+    >
+      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod
```

Comment:
If I make the content a `text` knob, it breaks the storybook. I assume there's some change or update lifecycle event I'm not listening for

Reply:
My best guess is that Storybook is trying to map the knob value as equal to `children`. Does it still break if you remove sibling elements (`<button>` etc.)?

Reply:
I have to use `dangerouslySetInnerHtml` in order to get the text to actually render as HTML, so I assume that's the issue. Low-priority at the moment :)